### PR TITLE
Fix accuracy logic - improves graphs

### PR DESCRIPTION
**Pull request by daeglin**

### CODE REVIEW

File: frontend/src/app/game/components/graphs/graphs.component.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export class GraphsComponent implements AfterViewInit {
       yAxes: [{
         ticks: {
           callback(value: number | string) {
-            return formatNumber(+value, false, true);
+            return formatNumber(+value, false, true, 2);
```
Comment (daeglin) [inline frontend/src/app/game/components/graphs/graphs.component.ts:28]: Force minimum two significant digits in graphs

Comment (daeglin) [reply frontend/src/app/game/components/graphs/graphs.component.ts:28]: Avoids problem with displaying many "0" s on the y axis

File: frontend/src/app/utils/format.ts [hunk 2]
```diff
@@ -1,25 +1,39 @@
 export function formatNumber(value: number, showCurrencySymbol = false, shrink = false,
-  accuracy = shrink ? 1 : 2): string {
+  accuracy = 1): string {
```
Comment (daeglin) [inline frontend/src/app/utils/format.ts:2]: For backward compatibility. Before this patch only percentage display overrides this explicitly to 1. So all numbers are effectively displayed like this

File: frontend/src/app/utils/format.ts [hunk 3]
```diff
@@ -1,25 +1,39 @@
...
+    let numDigitsBeforeDecimalPoint;
+    if (value === 0) {
+      numDigitsBeforeDecimalPoint = 1;  // 0 is a significant digit in this case
+      value = 0;                        // protect against -0
+    } else if (Math.abs(value) < 1) {
+      numDigitsBeforeDecimalPoint = 0;
+    } else {
+      numDigitsBeforeDecimalPoint = Math.floor(Math.log10(Math.abs(value))) + 1;
+    }
+    const numFractionDigits = Math.max(0, accuracy - numDigitsBeforeDecimalPoint)
```
Comment (daeglin) [inline frontend/src/app/utils/format.ts:32]: New logic: always display at least "accuracy" significant digits.

File: frontend/src/app/utils/format.ts [hunk 4]
```diff
@@ -1,25 +1,39 @@
...
       affix = ' mld.';
     }
-    else if (value >= 1_000_000) {
+    else if (Math.abs(value) >= 1_000_000) {
       value /= 1_000_000;
       affix = ' mil.';
     }
+
+    // Show at least one digit after decimal point when shrinking
+    if (affix !== '') {
```
Comment (daeglin) [inline frontend/src/app/utils/format.ts:17]: For at least two significant digits after shrinking

**Review by jnwltr — APPROVED** [review]

**Pull request merged**